tox.ini

```diff
@@ -64,17 +69,128 @@ allowlist_externals = env
 commands =
     env PATH="{env_bin_dir}" coverage run -m pytest {posargs:tests -ra --durations=20}
 
-[testenv:lint]
-base_python = py313,py312,py311,py310,py39
-use_develop = true
-deps =
-    pylint==3.2.7
-    pytest>=6
-    pytest-mock
-    importlib_metadata; python_version<"3.10"
-    iniconfig
+[testenv:pre-commit]
+description =
+  Run the quality checks under {basepython}; run as
+  `SKIP=check-id1,check-id2 tox r -e pre-commit` to instruct the underlying
+  `pre-commit` invocation avoid running said checks; Use
+  `tox r -e pre-commit -- check-id1 --all-files` to select checks matching IDs
+  aliases{:} `tox r -e pre-commit -- mypy --all-files` will run 3 MyPy
+  invocations, but `tox r -e pre-commit -- mypy-py313 --all-files` runs one.
 commands =
-    pylint {posargs:lektor tests}
+  {envpython} \
+    {[python-cli-options]byte-errors} \
+    {[python-cli-options]max-isolation} \
+    {[python-cli-options]warnings-to-errors} \
+    -m pre_commit \
```

Comment:
What does running python with all the paranoia options (-bb -E -s -i  -Werror -mI) buy us when running pre-commit?  Is it worth the verbosity?